File: src/main/scala/io/github/hexirp/akariki/Metadata.java
package io.github.hexirp.akariki;

import net.minecraft.util.ResourceLocation;

import org.apache.logging.log4j.core.Logger;

/**
 * {@link Metadata} is data of akariki.
 */
public class Metadata {
    private final String id;
    private final Logger log;

    public Metadata(String id, Logger log) {
        this.id = id;
        this.log = log;
    }

    public String id() {
        return id;
    }

    public ResourceLocation newResourceLocation(String name) {
        return new ResourceLocation(id, name);
    }

    public void info(String message) {
        log.info(message);
    }
}

File: src/main/scala/io/github/hexirp/akariki/Metadata.scala
package io.github.hexirp.akariki

import net.minecraft.util.ResourceLocation

import org.apache.logging.log4j.{LogManager, Logger}

/** [[Metadata]] は akariki に関するデータ群である。
  *
  * @param id      The MOD's ID
  * @param name    The MOD's name
  * @param version The MOD's version
  */
class Metadata(val id : String, val name : String, val version : String) {
  def newLogger : Logger = LogManager.getFormatterLogger(id)

  def newResourceLocation(resourceName : String) : ResourceLocation = new ResourceLocation(id, resourceName)
}
